File: in_Java/0922 Sort Array By Parity II.java
class Solution {
    public int[] sortArrayByParityII(int[] A) {
        int[] res = new int[A.length];
        int odd = 1;
        int even = 0;
        for(int i : A){
            if(i % 2 == 0){
                res[even] = i;
                even += 2;
            }else{
                res[odd] = i;
                odd += 2;
            }
        }
        return res;
    }
}

File: in_Python/0922 Sort Array By Parity II.py
class Solution(object):
    def sortArrayByParityII(self, A):
        """
        :type A: List[int]
        :rtype: List[int]
        """
        evens = []
        odds = []
        for a in A:
            if a % 2 == 0:
                evens.append(a)
            else:
                odds.append(a)
        res = []
        eid = 0
        oid = 0
        for i in range(len(A)):
            if i % 2 == 0:
                res.append(evens[eid])
                eid += 1
            else:
                res.append(odds[oid])
                oid += 1
        return res